services/frontend-service/src/ui/AppBar/LocksDrawer.tsx

```diff
@@ -116,75 +242,60 @@ export const LocksDrawer = (props: { data: GetOverviewResponse }) => {
+    const appLocks: LocksRow[] = [];
 
-        nwLocks.sort((a: Locks, b: Locks) => {
-            if (!a.commit?.authorTime || !b.commit?.authorTime) return 0;
-            if (a.commit.authorTime < b.commit.authorTime) return 1;
-            if (a.commit.authorTime === b.commit.authorTime) return 0;
-            return -1;
-        });
-        setLocks(nwLocks);
-    }, [data, calcAge]);
+    for (const env of Object.values(data.environments)) {
+        envLocks.push(
+            ...Object.entries(env.locks ?? {}).map((item) => ({
+                id: item[0],
+                author: item[1].commit?.authorName ?? '',
+                authorEmail: item[1].commit?.authorEmail ?? '',
+                message: item[1].message,
+                // use -1 to sort the locks with the newest on top
+                dateAdded: (item[1].commit?.authorTime?.valueOf() ?? -1) * -1,
+                environment: env.name,
+                application: '',
+            }))
+        );
+
+        for (const app of Object.values(env.applications)) {
+            appLocks.push(
+                ...Object.entries(app.locks ?? {}).map((item) => ({
+                    id: item[0],
+                    author: item[1].commit?.authorName ?? '',
+                    authorEmail: item[1].commit?.authorEmail ?? '',
+                    message: item[1].message,
+                    // use -1 to sort the locks with the newest on top
+                    dateAdded: (item[1].commit?.authorTime?.valueOf() ?? -1) * -1,
+                    environment: env.name,
+                    application: app.name,
+                }))
+            );
+        }
+    }
 
-    useEffect(() => {
-        setOutDatedLocks(locks.filter((lock) => !!lock.age && lock?.age > 2).length > 0);
-    }, [locks]);
+    const outdated =
+        envLocks.filter((lock) => isOutdated(lock.dateAdded)).length > 0 ||
+        appLocks.filter((lock) => isOutdated(lock.dateAdded)).length > 0;
 
     return (
         <>
             <Button
                 sx={{ color: theme.palette.grey[900], width: '100%' }}
                 variant={'contained'}
                 onClick={toggleDrawer(true)}>
-                <Badge
-                    invisible={!outDatedLocks}
-                    anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
-                    badgeContent={<WarningRoundedIcon color="error" />}>
-                    <strong>all locks</strong>
-                </Badge>
+                {outdated && <WarningRounded color="error" />}
+                <strong>Locks</strong>
                 <ExpandMoreRounded />
             </Button>
-            <Drawer anchor={'top'} open={state['isOpen']} onClose={toggleDrawer(false)}>
+            <Drawer anchor={'top'} open={drawerOpen} onClose={toggleDrawer(false)}>
                 <Box sx={{ width: 'auto' }} role="presentation">
-                    <AllLocks locks={locks} onClick={toggleDrawer(false)} />
```

Comment:
Assume we have a lot of looks, and the drawer takes the whole screen length, how we can close the drawer? did you handle this situation?

Reply:
It can be closed with `Esc` button. I think it could be useful to have some close button somewhere in the ui, but I haven't found a good way to do it yet. Do you think a ui close button would work?

Reply:
maybe create a small story for that?

Reply:
Yeah! Good idea. Thanks :)